kafka-filter-pypi/entrypoint.py

```diff
@@ -103,29 +130,38 @@ def _fill_old(self):
         )
         for message in consumer:
             pkg = message.value
-            if not self._exists(pkg):
+            if not self._exists_in_dictionary(pkg):
                 self._store(pkg)
 
     def _extract(self, package):
         try:
+            is_ingested = True if "ingested" in package else False
             pkg_name = package["project"]["info"]["name"]
             requires_dist = package["project"]["info"]["requires_dist"]
             releases = package["project"]["releases"]
         except KeyError:
             print ("Could not retrieve packaging info from {}".format(json.dumps(package)))
             return
-
+        if not requires_dist:
+            requires_dist = []
         requires_dist = self._parse_requires(requires_dist)
         for release in releases:
-            if not release.get("version", None) or\
-                    not release.get("releases", None):
-                continue
-
-            version = release["version"]
+            if is_ingested:
+                version = release
+                if not releases.get(version, None):
+                    continue
+                release_list = releases[version]
+            else:
+                if not release.get("version", None) or\
+                        not release.get("releases", None):
+                    continue
+                version = release["version"]
```

Comment:
In this case the `version` is `release["Version"]` while in the above `if` case `version` is `release`. Won't this lead to different outputs having different formats?

Reply:
I am not sure I fully understood this issue. Those 4 lines existed from before this PR, I just search with this heuristic only on non-ingested records (the ones which are scraped).

Reply:
Ah I see. Thanks for outlining.